packages/babel-plugin-proposal-pipeline-operator/src/minimalVisitor.js

```diff
@@ -0,0 +1,55 @@
+import { types as t } from "@babel/core";
+
+const minimalVisitor = {
+  BinaryExpression: {
+    exit(path) {
```

Comment:
Some of the calls in this function need `Expression` nodes and the smart proposal child nodes aren't expressions, but can be converted by the smart proposal visitor. Running on exit guarantees that the child nodes are already converted.

Reply:
This feels a bit like the logic is getting mixed between them. This might be a bit pendantic, but I'd extract the shared logic to a separate module and run them at different times.

Reply:
We could move this function to its own module and bind it to `entry` (or the default `BinaryExpression') on minimal and to `exit` on smart. But I'm not really sure about that. It feels like splitting hairs.

Plus: running on exit works great with the flattening PR that I've got in the works.